Sources/SecureXPC/XPCDecoder.swift

```diff
@@ -184,6 +184,22 @@ fileprivate class XPCSingleValueDecodingContainer: SingleValueDecodingContainer
     func decode(_ type: UInt64.Type) throws -> UInt64 {
         return try decode(xpcType: XPC_TYPE_UINT64, transform: xpc_uint64_get_value)
     }
+
+	func decodeXPCConnection() throws -> xpc_connection_t {
+		return try decode(xpcType: XPC_TYPE_ENDPOINT, transform: { $0 })
+	}
+
+	func decodeXPCEndPoint() throws -> xpc_endpoint_t {
+		return try decode(xpcType: XPC_TYPE_ENDPOINT, transform: xpc_connection_create_from_endpoint)
+	}
+
+//	func decode(_ type: xpc_connection_t.Type) throws -> xpc_connection_t {
+//		return try decode(xpcType: XPC_TYPE_ENDPOINT, transform: { $0 })
+//	}
+//
+//	func decode(_ type: xpc_endpoint_t.Type) throws -> xpc_endpoint_t {
+//		return try decode(xpcType: XPC_TYPE_ENDPOINT, transform: xpc_connection_create_from_endpoint)
+//	}
```

Comment:
These spellings aren't legal, because `xpc_connection_t` and `xpc_endpoint_t` lower to the same type.

*flashbacks to Java's jank generics 😖*